backend/ticketvise/views/api/security.py

```diff
@@ -26,116 +25,95 @@ def has_permission(self, request, view):
         return True
 
 
-# class UserIsInInboxMixin(AccessMixin):
-#     inbox_key = "inbox_id"
-#
-#     def dispatch(self, request, *args, **kwargs):
-#         if not request.user.is_authenticated:
-#             return self.handle_no_permission()
-#
-#         if request.user.is_superuser:
-#             return super().dispatch(request, *args, **kwargs)
-#
-#         inbox_id = kwargs.get(self.inbox_key)
-#         if not inbox_id:
-#             return self.handle_no_permission()
-#
-#         inbox = get_object_or_404(Inbox, pk=inbox_id)
-#         if not request.user.has_inbox(inbox) and not request.user.is_assistant_or_coordinator(inbox):
-#             return self.handle_no_permission()
-#
-#         return super().dispatch(request, *args, **kwargs)
-
-
-class UserIsInboxStaffMixin(AccessMixin):
-    inbox_key = "inbox_id"
+class UserIsInboxStaffPermission(IsAuthenticated):
 
-    def dispatch(self, request, *args, **kwargs):
+    def has_permission(self, request, view):
         if not request.user.is_authenticated:
-            return self.handle_no_permission()
+            return False
 
-        inbox_id = kwargs.get(self.inbox_key)
+        if request.user.is_superuser:
+            return True
+
+        inbox_id = view.kwargs.get("inbox_id")
         if not inbox_id:
-            return self.handle_no_permission()
+            return False
 
         inbox = get_object_or_404(Inbox, pk=inbox_id)
         if not request.user.is_assistant_or_coordinator(inbox):
-            return self.handle_no_permission()
+            return False
 
-        return super().dispatch(request, *args, **kwargs)
+        return True
 
 
-class UserIsInboxManagerMixin(AccessMixin):
-    inbox_key = "inbox_id"
+class UserIsInboxManagerPermission(IsAuthenticated):
 
-    def dispatch(self, request, *args, **kwargs):
+    def has_permission(self, request, view):
         if not request.user.is_authenticated:
-            return self.handle_no_permission()
+            return False
 
-        inbox_id = kwargs.get(self.inbox_key)
+        inbox_id = view.kwargs.get("inbox_id")
         if not inbox_id:
-            return self.handle_no_permission()
+            return False
 
         inbox = get_object_or_404(Inbox, pk=inbox_id)
         if not request.user.is_coordinator_for_inbox(inbox):
-            return self.handle_no_permission()
+            return False
 
-        return super().dispatch(request, *args, **kwargs)
+        return True
```

Comment:
Defensive programming when it comes to permissions could save quite a bit of headache down the road. 

Return False by default except if some conditions are met, instead of the other way around. This way it's less likely a permission will be granted by accident (better to deny something and receive some questions, than to lose data).

Reply:
Thanks for the heads up!